TACC/Core-Portal
# task/TV3-168  Tapis Client UI Delay

## Overview

Quick change to move the tapis client refresh_tokens() call out of the middleware. The current issue we are seeing is that in instances where a user token expires, it is taking ~15 seconds to refresh tokens which is causing elements on the initial portal landing page to not load. With this change we are no longer intercepting and making tapis calls when non tapis related actions/calls are taking place, such as during the initial portal landing page load. The longer refresh token time still exists (cic team is investigating this)

It is also difficult to recreate this issue locally since this longer refresh token time only seems to pop up when a user token naturally expires (not when we manually expire it) 

## Related

* [TV3-168](https://jira.tacc.utexas.edu/browse/TV3-168)

## Changes

- Moved refresh_tokens() call out of middleware and put it along with the tapis client initialization

## Testing

1. Run the portal as normal and make sure functionality is working
2. Manually expire your auth token
3. Refresh the portal and ensure the functionality continues to work


## Notes
